src/Driver/OCI8/Connection.php

```diff
@@ -47,22 +47,29 @@ public function getServerVersion(): string
         return $matches[1];
     }
 
-    /** @throws Parser\Exception */
+    /**
+     * @throws Parser\Exception
+     * @throws Error
+     */
     public function prepare(string $sql): Statement
     {
         $visitor = new ConvertPositionalToNamedPlaceholders();
 
         $this->parser->parse($sql, $visitor);
 
-        $statement = oci_parse($this->connection, $visitor->getSQL());
-        assert(is_resource($statement));
+        $statement = @oci_parse($this->connection, $visitor->getSQL());
+
+        if (! is_resource($statement)) {
+            throw Error::new($this->connection);
+        }
 
         return new Statement($this->connection, $statement, $visitor->getParameterMap(), $this->executionMode);
     }
 
     /**
      * @throws Exception
      * @throws Parser\Exception
+     * @throws Error
```

Comment:
`Error` is a subtype of `Exception`, so it's already declared.

Reply:
Done